pilot/pkg/serviceregistry/kube/controller/network.go

```diff
@@ -81,8 +81,10 @@ func initNetworkManager(c *Controller, options Options) *networkManager {
 		gatewaysFromResource:           make(map[types.UID]model.NetworkGatewaySet),
 		discoverRemoteGatewayResources: options.ConfigCluster,
 	}
+	// always initialize the gatewayResourceClient since ambient index will use it to configure waypoints
+	n.gatewayResourceClient = kclient.NewDelayedInformer[*v1beta1.Gateway](c.client, gvr.KubernetesGateway, kubetypes.StandardInformer, kubetypes.Filter{})
```

Comment:
@howardjohn, I think this is actually too aggressive. We should probably initialize `if features.MultiNetworkGatewayAPI || features.EnableAmbientControllers`. Would you like me to push that?

Reply:
Yep. I have this done in https://github.com/istio/istio/pull/46501